[ppc64le] test suite failing on power9

As suggested by  @takluyver, here are the results of the test-suite running on IBM power9 computer running Ubuntu 18.04. The python core is 3.6, all modules have been pip installed in a venv (and recompiled as no binary wheels exist yet on this platform) and HDF5 has been compile from sources in version 1.10.5.

Most failing tests are related in a way or another to "float128" which have just been deactivated in  #1243. They should probably be marked as "xfailed".

```
(venv-kieffer_system) test@power9:~/HDF5/h5py/build/lib.linux-ppc64le-3.6$ python
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import h5py
>>> h5py.run_tests()
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.6.7, pytest-4.4.2, py-1.8.0, pluggy-0.11.0
rootdir: /home/test/HDF5/h5py, inifile: pytest.ini
collected 537 items                                                                                                                                                                       

h5py/tests/hl/test_attribute_create.py ...                                                                                                                                          [  0%]
h5py/tests/hl/test_completions.py ..                                                                                                                                                [  0%]
h5py/tests/hl/test_dataset_getitem.py .................................................x....................................                                                        [ 16%]
h5py/tests/hl/test_dataset_swmr.py ssss..........                                                                                                                                   [ 19%]
h5py/tests/hl/test_datatype.py .........F.....ssss                                                                                                                                  [ 23%]
h5py/tests/hl/test_deprecation.py .                                                                                                                                                 [ 23%]
h5py/tests/hl/test_dims_dimensionproxy.py .                                                                                                                                         [ 23%]
h5py/tests/hl/test_file.py .................                                                                                                                                        [ 26%]
h5py/tests/hl/test_filters.py .s.                                                                                                                                                   [ 27%]
h5py/tests/hl/test_threads.py ..                                                                                                                                                    [ 27%]
h5py/tests/hl/test_vds/test_highlevel_vds.py ......                                                                                                                                 [ 28%]
h5py/tests/hl/test_vds/test_lowlevel_vds.py ...                                                                                                                                     [ 29%]
h5py/tests/hl/test_vds/test_virtual_source.py ................                                                                                                                      [ 32%]
h5py/tests/old/test_attrs.py ...............s                                                                                                                                       [ 35%]
h5py/tests/old/test_attrs_data.py ......F.............                                                                                                                              [ 38%]
h5py/tests/old/test_base.py .....                                                                                                                                                   [ 39%]
h5py/tests/old/test_dataset.py ......F..x...............................s.................................x..........................                                               [ 58%]
h5py/tests/old/test_datatype.py ..                                                                                                                                                  [ 59%]
h5py/tests/old/test_dimension_scales.py ....s................                                                                                                                       [ 63%]
h5py/tests/old/test_file.py .............ss...ssss....................s.........                                                                                                    [ 72%]
h5py/tests/old/test_file_image.py ..                                                                                                                                                [ 73%]
h5py/tests/old/test_group.py .................................ssssss...................................................                                                             [ 89%]
h5py/tests/old/test_h5.py .....                                                                                                                                                     [ 90%]
h5py/tests/old/test_h5d_direct_chunk_write.py .                                                                                                                                     [ 91%]
h5py/tests/old/test_h5f.py .....                                                                                                                                                    [ 91%]
h5py/tests/old/test_h5p.py ........                                                                                                                                                 [ 93%]
h5py/tests/old/test_h5t.py ....                                                                                                                                                     [ 94%]
h5py/tests/old/test_objects.py ...                                                                                                                                                  [ 94%]
h5py/tests/old/test_selections.py ....                                                                                                                                              [ 95%]
h5py/tests/old/test_slicing.py ........................                                                                                                                             [100%]

======================================================================================== FAILURES =========================================================================================
__________________________________________________________________________ TestOffsets.test_float_round_tripping __________________________________________________________________________

self = <h5py.tests.hl.test_datatype.TestOffsets testMethod=test_float_round_tripping>

    @ut.skipIf(
        platform.machine() in x86_32_BIT_SYSTEMS,
        'Test fails on i386, need to sort out long double FIX THIS')
    def test_float_round_tripping(self):
        dtypes = set(f for f in np.typeDict.values()
                     if (np.issubdtype(f, np.floating) or
                         np.issubdtype(f, np.complexfloating))
                     )
    
        dtype_dset_map = {str(j): d
                          for j, d in enumerate(dtypes)}
    
        fname = self.mktemp()
    
        with h5py.File(fname, 'w') as f:
            for n, d in dtype_dset_map.items():
                data = np.arange(10,
                                 dtype=d)
    
                f.create_dataset(n, data=data)
    
        with h5py.File(fname, 'r') as f:
            for n, d in dtype_dset_map.items():
                ldata = f[n][:]
>               self.assertEqual(ldata.dtype, d)
E               AssertionError: dtype('<f8') != <class 'numpy.float128'>

h5py/tests/hl/test_datatype.py:307: AssertionError
__________________________________________________________________________________ TestTypes.test_float ___________________________________________________________________________________

self = <h5py.tests.old.test_attrs_data.TestTypes testMethod=test_float>

    def test_float(self):
        """ Storage of floating point types """
        dtypes = tuple(np.dtype(x) for x in ('<f4','>f4','<f8','>f8'))
    
        for dt in dtypes:
            data = np.ndarray((1,), dtype=dt)
            data[...] = 42.3
            self.f.attrs['x'] = data
            out = self.f.attrs['x']
>           self.assertEqual(out.dtype, dt)
E           AssertionError: dtype('>f16') != dtype('>f8')

h5py/tests/old/test_attrs_data.py:112: AssertionError
____________________________________________________________________________ TestCreateShape.test_long_double _____________________________________________________________________________

self = <h5py.tests.old.test_dataset.TestCreateShape testMethod=test_long_double>

    def test_long_double(self):
        """ Confirm that the default dtype is float """
        dset = self.f.create_dataset('foo', (63,), dtype=np.longdouble)
>       self.assertEqual(dset.dtype, np.longdouble)
E       AssertionError: dtype('<f8') != <class 'numpy.float128'>

h5py/tests/old/test_dataset.py:94: AssertionError
==================================================================================== warnings summary =====================================================================================
build/lib.linux-ppc64le-3.6/h5py/tests/hl/test_dataset_getitem.py::Test1DZeroFloat::test_mask
  /home/test/HDF5/h5py/build/lib.linux-ppc64le-3.6/h5py/_hl/dataset.py:534: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
    if args == (Ellipsis,) or args == tuple():

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================= 3 failed, 506 passed, 25 skipped, 3 xfailed, 1 warnings in 3.97 seconds =========================================================
1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -66,6 +66,13 @@ matrix:
         HDF5_VERSION=1.10.5
         HDF5_DIR=$HDF5_CACHE_DIR/$HDF5_VERSION
 
+    - python: 3.7
+      env:
+      - TOXENV=py37-test-deps
+        HDF5_VERSION=1.10.5
+        HDF5_DIR=$HDF5_CACHE_DIR/$HDF5_VERSION
+      os: linux-ppc64le
+
     # MPI tests
     # TODO: We should test with newer versions of HDF5
     - python: 3.6

--- a/h5py/h5t.pyx
+++ b/h5py/h5t.pyx
@@ -1371,23 +1371,21 @@ def _get_float_dtype_to_hdf5():
     float_be = {}
     h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE, LDOUBLE_BE]
     h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE, LDOUBLE_LE]
-    
+ 
     for ftype_, finfo, size in _available_ftypes:
         nmant, maxexp, minexp = _correct_float_info(ftype_, finfo)
         for h5type in h5_be_list:
             spos, epos, esize, mpos, msize = h5type.get_fields()
             ebias = h5type.get_ebias()
             if (finfo.iexp == esize and nmant == msize and
-                (maxexp - 1) == ebias
-            ):
+                    (maxexp - 1) == ebias):
                 float_be[ftype_] = h5type
                 break # first found matches, related to #1244
         for h5type in h5_le_list:
             spos, epos, esize, mpos, msize = h5type.get_fields()
             ebias = h5type.get_ebias()
             if (finfo.iexp == esize and nmant == msize and
-                (maxexp - 1) == ebias
-            ):
+                    (maxexp - 1) == ebias):
                 float_le[ftype_] = h5type
                 break # first found matches, related to #1244
     if ORDER_NATIVE == H5T_ORDER_LE:

--- a/h5py/h5t.pyx
+++ b/h5py/h5t.pyx
@@ -1369,11 +1369,9 @@ cdef class TypeEnumID(TypeCompositeID):
 def _get_float_dtype_to_hdf5():
     float_le = {}
     float_be = {}
-    h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE,
-                  LDOUBLE_BE]
-    h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE]
-    if MACHINE != 'ppc64le':
-        h5_le_list.append(LDOUBLE_LE)
+    h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE, LDOUBLE_BE]
+    h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE, LDOUBLE_LE]
+    
     for ftype_, finfo, size in _available_ftypes:
         nmant, maxexp, minexp = _correct_float_info(ftype_, finfo)
         for h5type in h5_be_list:
@@ -1383,6 +1381,7 @@ def _get_float_dtype_to_hdf5():
                 (maxexp - 1) == ebias
             ):
                 float_be[ftype_] = h5type
+                break # first found matches, related to #1244
         for h5type in h5_le_list:
             spos, epos, esize, mpos, msize = h5type.get_fields()
             ebias = h5type.get_ebias()
@@ -1390,6 +1389,7 @@ def _get_float_dtype_to_hdf5():
                 (maxexp - 1) == ebias
             ):
                 float_le[ftype_] = h5type
+                break # first found matches, related to #1244
     if ORDER_NATIVE == H5T_ORDER_LE:
         float_nt = dict(float_le)
     else:

--- a/h5py/h5t.pyx
+++ b/h5py/h5t.pyx
@@ -1371,7 +1371,7 @@ def _get_float_dtype_to_hdf5():
     float_be = {}
     h5_be_list = [IEEE_F16BE, IEEE_F32BE, IEEE_F64BE, IEEE_F128BE, LDOUBLE_BE]
     h5_le_list = [IEEE_F16LE, IEEE_F32LE, IEEE_F64LE, IEEE_F128LE, LDOUBLE_LE]
- 
+
     for ftype_, finfo, size in _available_ftypes:
         nmant, maxexp, minexp = _correct_float_info(ftype_, finfo)
         for h5type in h5_be_list:

